addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -299,7 +299,9 @@ const Wysiwyg = Widget.extend({
         this.$editable.on('click', '.o_image, .media_iframe_video', e => e.preventDefault());
         this.showTooltip = true;
         this.$editable.on('dblclick', mediaSelector, function () {
-            if (this.isContentEditable || (this.parentElement && this.parentElement.isContentEditable)) {
+            if (this.isContentEditable 
+                    || (this.parentElement && this.parentElement.isContentEditable
+                    || this.classList.contains('o_editable_media'))) {
```

Comment:
I'm so confused, on runbot v15.0 standard (non-custom) social media icons _can_ be replaced, both by double-click and with the replace button. I believe that wasn't the case locally when I forked. More importantly, I thought it should not be the case ?

Reply:
What I do notice on v16 is that
(1) if I enter edit mode and directly click on "replace" for a custom social media icon, it doesn't work. It pretty much only works once you've found another way to open the media dialog. It's the selector issue whose fix you didn't want in this PR @dmo-odoo.
(2) If the social media snippet and its custom icon are manually added (they're not the default ones) in this "edit session" (since clicking edit), then clicking replace does work. And if before saving during this first edit session you click on replace, all subsequent sessions will work even from the beginning. But if you add it and save it without clicking on replace, then all subsequent edit sessions will face the same issue (1) when clicking on replace.
Funny, isn't it ? Just thought I'd document something from all these test sessions.

Reply:
> Right now, all social media icons that get the o_editable_media class can be replaced by double-clicking.

The original specs were that double clicking can/should allow replacing the image but we should not display the "replace" button. The logic being "it does not really make sense to change the icon for a specific social media so we should not advertise it but it's ok/good if we can do it via double-click". When the feature was merged, I think that's the behavior we had too.

So if you say "all media with o_editable_media are replaceable via double click", it seems good. As for the replace button I thought it was already targeting custom ones only explicitely but I may be wrong:

> Wherever is the code that decides the "Replace" button appears or not

You can check `ReplaceMedia`, its `data-selector` and/or its `_computeWidgetVisibility` function.